go/pkg/filemetadata/filemetadata_test.go

```diff
@@ -64,7 +78,62 @@ func TestComputeFiles(t *testing.T) {
 	}
 }
 
+func TestComputeFilesWithXattr(t *testing.T) {
+	XattrDigestName = "google.digest.sha256"
+	XattrAccess = xattributeAccessorMock{}
+	tests := []struct {
+		name       string
+		contents   string
+		executable bool
+	}{
+		{
+			name:     "empty",
+			contents: "",
+		},
+		{
+			name:     "non-executable",
+			contents: "bla",
+		},
+		{
+			name:       "executable",
+			contents:   "foo",
+			executable: true,
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			getXAttrMock = func(_ string, _ string) ([]byte, error) {
+				return []byte(tc.name), nil
+			}
+
+			before := time.Now().Truncate(time.Second)
+			filename, err := testutil.CreateFile(t, tc.executable, tc.contents)
+			if err != nil {
+				t.Fatalf("Failed to create tmp file for testing digests: %v", err)
+			}
+			after := time.Now().Truncate(time.Second).Add(time.Second)
+			defer os.RemoveAll(filename)
+			got := Compute(filename)
+			if got.Err != nil {
+				t.Errorf("Compute(%v) failed. Got error: %v", filename, got.Err)
+			}
+			wantDigest, _ := digest.NewFromString(fmt.Sprintf("%s/%d", tc.name, len(tc.contents)))
+			want := &Metadata{
+				Digest:       wantDigest,
+				IsExecutable: tc.executable,
+			}
+			if diff := cmp.Diff(want, got, ignoreMtime); diff != "" {
+				t.Errorf("Compute(%v) returned diff. (-want +got)\n%s", filename, diff)
+			}
+			if got.MTime.Before(before) || got.MTime.After(after) {
+				t.Errorf("Compute(%v) returned MTime %v, want time in (%v, %v).", filename, got.MTime, before, after)
+			}
+		})
+	}
+}
+
 func TestComputeDirectory(t *testing.T) {
+	XattrDigestName = ""
```

Comment:
Yeah, that was the issue I was trying to avoid: the Compute() function would need to be changed which could have consequences; that's why I was going to go with the suggestion to just make it a const.  But then I realized the Mock is in the same situation anyway, can't be done in parallel because there's no way to pass extra data through the Compute() function without changing the API.  My logic is that since it needs to be that way to properly unit test anyway, and all tests are already sequential anyway, there's no (current) downside to doing it this way, and plenty of upside.  If we want to optimize the tests in the future to run in parallel the API can be modified then.

Reply:
Sgtm to do it when we decide on optimizing the test.